src/solc_0.8/asset/AssetV2.sol

```diff
@@ -6,5 +6,19 @@ import "./ERC1155ERC721.sol";
 
 // solhint-disable-next-line no-empty-blocks
 contract AssetV2 is ERC1155ERC721 {
-
+    // polygon-specific layer 1 functionality
+    function mintBatch(
+        address to,
+        uint256[] calldata ids,
+        uint256[] calldata amounts,
+        bytes calldata data
+    ) external {
+        // @review
+        require(_msgSender() == _predicate, "!PREDICATE");
+        // operator here could be predicate... is there a benefit?
+        // hash should be the bytes32 metaData-hash.
+        bytes32 dummyHash = bytes32("0x00");
+        address operator = address(uint160(0));
+        _mint(dummyHash, amounts[0], 0, operator, to, ids[0], data, false);
```

Comment:
Need to think about how we would set the hash in the call to `_mint(` here if the predicate does not pass the data to `mintBatch(...)`.